dask/dataframe/core.py

```diff
@@ -2415,6 +2415,14 @@ def to_series(self):
         return self.map_partitions(M.to_series,
                                    meta=self._meta.to_series())
 
+    @derived_from(pd.Index)
+    def to_frame(self, name=None):
```

Comment:
According to the [`pandas.Index.to_frame` API doc](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.to_frame.html) it looks like there should also be a `index=` keyword here

Reply:
It's true, but I think it's not possible to implement this in dask since each partition will have it's own index.

Reply:
What do you mean @henriqueribeiro? Wouldn't it have similar behavior as dask dataframe's `.reset_index`?

Reply:
I will give an example as it's easier.

```python
import numpy as np
import pandas as pd
import dask.dataframe as dd

d = {'index': range(10),
     'data': np.random.normal(size=10)
    }
df = pd.DataFrame(d).set_index('index')
ddf = dd.from_pandas(df, npartitions=2)
```
```python
df.index.to_frame(False)
```
|    |   index |
|---:|--------:|
|  0 |       0 |
|  1 |       1 |
|  2 |       2 |
|  3 |       3 |
|  4 |       4 |
|  5 |       5 |
|  6 |       6 |
|  7 |       7 |
|  8 |       8 |
|  9 |       9 |

```python
ddf.index.to_frame(False).compute()
```
|    |   index |
|---:|--------:|
|  0 |       0 |
|  1 |       1 |
|  2 |       2 |
|  3 |       3 |
|  4 |       4 |
|  0 |       5 |
|  1 |       6 |
|  2 |       7 |
|  3 |       8 |
|  4 |       9 |

As you can see, since the dask dataframe has 2 partitions the index "restarts" in the middle. There is a way we can avoid this?

Reply:
No, it's unavoidable. Notice how `reset_index` is the same

```python
In [5]: ddf.reset_index(drop=True).compute()
Out[5]:
       data
0 -0.868215
1 -1.713328
2  0.050752
3 -0.206222
4  0.545437
0  0.494061
1  0.567893
2  0.475889
3  1.460188
4  0.414024
```

Reply:
@henriqueribeiro could you please add an `index=` keyword here that is `True` by default and raise an error if `index` isn't `True`. That way we will match the corresponding pandas `to_frame` signature and avoid situations like

```python
ddf.index.to_frame(False)
```

which will currently work here, but may behave unexpectedly to someone coming from pandas. 

As a side note, the `derived_from` decorator has a `ua_args` that is a list of keywords that are unsupported in Dask (e.g. `ua_args=['index']`).